src/Button/PrintButton/PrintButton.tsx

```diff
@@ -158,32 +177,47 @@ const PrintButton: React.FC<PrintButtonProps> = ({
           // second page for legends
           doc.addPage('a4', 'l');
 
-          pdfSpec.layers.forEach((layer: any, idx) => {
+          // add a legend title
+          doc.setFont('arial', 'bold');
+          doc.setFontSize(20);
+          doc.text(legendTitle, 148.5, 13, { align: 'center' });
+
+          let xPosition = 20;
+
+          for (let i = 0; i < pdfSpec.layers?.length; i++) {
+            const layer: any = pdfSpec.layers?.[i];
             const legendUrl = layer.legendUrl;
             const name = layer.layerName;
-            const xPosition = 20 + (60 * idx); // todo: dynamic size
+
             if (legendUrl && name) {
-              // todo: use actual legend image size
-              doc.setFont('arial', 'bold');
-              doc.setFontSize(12);
-              doc.text(name, xPosition, 17, { align: 'left' });
-              doc.addImage(legendUrl, 'PNG', xPosition, 20, 60, 60);
+              try {
+                // determine legend image dimensions
+                const response = await fetch(legendUrl);
+                const blob = await response.blob();
+                const base64 = await readBlob(blob);
```

Comment:
Untested, but this might be simplified to `const base64 = URL.createObjectURL(blob);`

Reply:
nice, that also works!

Reply:
added in 08b46ed859cc7314bdb0ecf098121e9fbdc7ab80